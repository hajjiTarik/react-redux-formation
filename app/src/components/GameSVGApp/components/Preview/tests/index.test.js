import React from "react";
import renderer from 'react-test-renderer';
import { Preview } from "../index";

function renderWithMutations (Component, mutations)  {
  mutations.map((mutation)=> {
    it(mutation.test, ()=> {
      const component = renderer.create(<Component {...mutation.props} />).toJSON();
      expect(component).toMatchSnapshot();
    });
  })
}

describe("Test Preview", ()=> {
    const mutations = [
      {
        test: "Should render without props",
        props: {
          rotate: "bobo"
        }
      },
      {
        test: "Should render without props",
        props: {
          rotate: "tootoo"
        }
      }
    ];
    renderWithMutations(Preview, mutations);
});
