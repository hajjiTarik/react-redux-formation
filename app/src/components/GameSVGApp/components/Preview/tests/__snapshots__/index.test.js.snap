// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Preview Should render without props 1`] = `
<div
  className="preview-container"
  style={
    Object {
      "backgroundColor": "transparent",
      "backgroundImage": "linear-gradient(transparent, transparent)",
    }
  }
>
  <img
    alt="error"
    style={
      Object {
        "transform": "rotate(bobodeg)",
      }
    }
  />
</div>
`;

exports[`Test Preview Should render without props 2`] = `
<div
  className="preview-container"
  style={
    Object {
      "backgroundColor": "transparent",
      "backgroundImage": "linear-gradient(transparent, transparent)",
    }
  }
>
  <img
    alt="error"
    style={
      Object {
        "transform": "rotate(tootoodeg)",
      }
    }
  />
</div>
`;
